node/src/components/block_validator.rs

```diff
@@ -98,18 +112,195 @@ pub(crate) enum Event {
     /// Deploy was invalid. Unable to convert to a deploy type.
     #[display(fmt = "{} invalid", _0)]
     CannotConvertDeploy(DeployOrTransferHash),
+
+    /// A finality signature was correctly fetched.
+    #[display(fmt = "{} found", _1)]
+    SignatureFound(ProposedBlock<ClContext>, Box<FinalitySignatureId>),
+
+    /// A finality signature couldn't be found or was invalid.
+    #[display(fmt = "{} invalid or missing", _1)]
+    SignatureError(ProposedBlock<ClContext>, Box<FinalitySignatureId>),
+}
+
+/// The state of the process of validating cited finality signatures.
+#[derive(DataSize, Debug)]
+enum SignaturesValidationState {
```

Comment:
Can you consider adding a graph describing the state transitions?
Like we did for some of the 1.5 state machines, for example [here](https://github.com/casper-network/casper-node/blob/dev/node/src/components.rs#L81-L92).
https://mermaid.live/ can come in handy.

Reply:
I'll do it in a separate PR if that's okay - it would take me a bit longer, and I'm eager to merge this PR ASAP.

Reply:
Sure, definitely not a blocker!